client/packages/system/src/Asset/api/hooks/log/useAssetLogReasons.ts

```diff
@@ -0,0 +1,18 @@
+import { useQuery, useUrlQueryParams } from '@openmsupply-client/common';
+import { useAssetApi } from '../utils/useAssetApi';
+
+export const useAssetLogReasons = (storeId?: string | undefined) => {
```

Comment:
The list of reasons is not showing a filtered list based on the status supplied.
To fix this I've changed to

```
export const useAssetLogReasons = (filter?: FilterByWithBoolean) => {
  const { queryParams } = useUrlQueryParams({
    filters: [
      { key: 'id' },
      { key: 'assetLogStatus', condition: 'equalTo' },
      { key: 'reason' },
    ],
  });

  const filterBy = filter ?? queryParams.filterBy ?? undefined;
  const api = useAssetApi();

  return useQuery(api.keys.logReasonsList({ ...queryParams, filterBy }), () =>
    api.get.logReasons(filterBy)
  );
};
```

and updated `StatusTab` to 

```
  const { data } = useAssetData.log.listReasons(
    draft.status
      ? {
          assetLogStatus: { equalTo: draft.status },
        }
      : undefined
  );
```